hits_cpp/shufflesGenerator.hpp

```diff
@@ -13,24 +13,36 @@ class MemeShuffler {
             maxShuffles = len;
         }
         this->_maxShuffles = maxShuffles;
+        this->_lenPatterns=len;
+        cout << "len pattern: " << this->_lenPatterns << endl;
+        this->_numPattern=-1;
```

Comment:
Constant member should be initialized before constructor body.
see how `_source`, `_patterns` and `_currentShuffle` are initialized

Reply:
Still not fixed, `_numPatterns` should also be initialized at start of construction, see `_currentShuffle`